scalr/helpers.go

```diff
@@ -12,25 +12,47 @@ func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
-func GetEnvironmentByName(environmentName string, scalrClient *scalr.Client) (*scalr.Environment, error) {
-	var environment *scalr.Environment
+type GetEnvironmentByNameOptions struct {
+	Name    *string
+	Account *string `json:",omitempty"`
+	Include *string `json:",omitempty"`
+}
 
-	envl, err := scalrClient.Environments.List(ctx)
+func GetEnvironmentByName(options GetEnvironmentByNameOptions, scalrClient *scalr.Client) (*scalr.Environment, error) {
+	listOptions := scalr.EnvironmentListOptions{
+		Name:    options.Name,
+		Account: options.Account,
+		Include: options.Include,
+	}
+	envl, err := scalrClient.Environments.List(ctx, listOptions)
 	if err != nil {
 		return nil, fmt.Errorf("Error retrieving environments: %v", err)
 	}
 
+	if len(envl.Items) == 0 {
+		return nil, fmt.Errorf("Environment with name '%s' not found or user unauthorized", *options.Name)
+	}
+
+	var matchedEnvironments []*scalr.Environment
+
+	// filter in endpoint search environments that contains quering string, this is why we need to do exeact match on our side.
 	for _, env := range envl.Items {
-		if env.Name == environmentName {
-			environment = env
-			break
+		if env.Name == *options.Name {
+			matchedEnvironments = append(matchedEnvironments, env)
 		}
 	}
-	if environment == nil {
-		return nil, fmt.Errorf("Could not find environment with name: %s", environmentName)
-	}
 
-	return environment, nil
+	switch numberOfMatch := len(matchedEnvironments); {
```

Comment:
Strange `switch case`. In theory, `numberOfMatch` can not be equal to zero, this case handles code above.

Reply:
This situation is possible when we have for example 3 environments with names `name1`, `name2`, and `name3`. For example, we are looking for an environment with the name `name`. Because of the implementation of the filter on the backend side, we will receive all those environments but neither of them is the right one.